src/test/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbeTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022 Jenkins Infra
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.jenkins.pluginhealth.scoring.probes;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.time.ZonedDateTime;
+
+import io.jenkins.pluginhealth.scoring.config.GithubConfiguration;
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+import io.jenkins.pluginhealth.scoring.model.ResultStatus;
+
+import org.junit.jupiter.api.Test;
+
+class SCMLinkValidationProbeTest {
+    @Test
+    public void shouldKeepScmAsKey() {
+        assertThat(new SCMLinkValidationProbe(new GithubConfiguration("")).key()).isEqualTo("scm");
+    }
+
+    @Test
+    public void shouldRequireRelease() {
+        assertThat(new SCMLinkValidationProbe(new GithubConfiguration("")).requiresRelease()).isTrue();
+    }
+
+    @Test
+    public void shouldNotAcceptNullNorEmptyScm() {
+        final Plugin p1 = new Plugin("foo", null, ZonedDateTime.now());
+        final Plugin p2 = new Plugin("bar", "", ZonedDateTime.now());
+        final SCMLinkValidationProbe probe = new SCMLinkValidationProbe(new GithubConfiguration(""));
```

Comment:
Here, you're using an empty `oauth` String, why?
[Correct me if I'm wrong]: I guess, maybe because the following tests are just checking the scm URL's validity using the regex expression which doesn't require us to contact the server. Would we need to write any test which would require us to use a valid `oauth`? If yes, can you give one example?